app/controllers/flowr_controller.rb

```diff
@@ -0,0 +1,42 @@
+class FlowrController < ApplicationController
+
+  def insights
+    unless current_user
+      skip_authorization
```

Comment:
What about using pundit for authorization management here? If we do so, we could remove lines 4-10.